src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -1328,6 +1328,12 @@ class G1MergeHeapRootsTask : public WorkerTask {
 
   ~G1MergeHeapRootsTask() {
     if (_dirty_card_buffers != nullptr) {
+#ifdef ASSERT
+      using Stack = BufferNode::Stack;
+      for (uint i = 0; i < _num_workers; i++) {
+        _dirty_card_buffers[i].~Stack();
+      }
+#endif
```

Comment:
I do not think this should be put under `#ifdef` as the destructor itself is always there afair. The compiler should automatically compile the whole loop away in product mode anyway.